nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_in_allowed_area.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_IN_ALLOWED_AREA_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_IN_ALLOWED_AREA_HPP_
+
+#include <string>
+#include <atomic>
+#include <deque>
+
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+#include "std_msgs/msg/bool.hpp"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief A BT::ConditionNode that subscribes to cart_in_allowed_area topic and returns the result
+ */
+class IsInAllowedArea : public BT::ConditionNode
+{
+public:
+  /**
+   * @brief A constructor for nav2_behavior_tree::IsInAllowedArea
+   * @param condition_name Name for the XML tag for this node
+   * @param conf BT node configuration
+   */
+  IsInAllowedArea(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf);
+
+  IsInAllowedArea() = delete;
+
+  /**
+   * @brief A destructor for nav2_behavior_tree::IsInAllowedArea
+   */
+  ~IsInAllowedArea() override;
```

Comment:
Why is this destructor marked as `override`. I don't think the base `ConditionNode` class has a `~IsInAllowedArea` destructor that needs overriding.